File: backend/src/main/java/project/backend/global/util/EmptyToNullDeserializer.java
package project.backend.global.util;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;
import java.io.IOException;

public class EmptyToNullDeserializer extends JsonDeserializer<String> {

	@Override
	public String deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
		String value = p.getValueAsString();
		return (value == null || value.isBlank()) ? null : value;
	}
}

File: backend/src/main/java/project/backend/global/util/EmptyToNullDeserializer.kt
package project.backend.global.util

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer

class EmptyToNullDeserializer : JsonDeserializer<String>() {
    override fun deserialize(p: JsonParser, ctxt: DeserializationContext): String? {
        val value = p.valueAsString
        return if ((value == null || value.isBlank())) null else value
    }
}